manifests/custom_service.pp

```diff
@@ -45,30 +43,26 @@
   Enum['present','absent'] $ensure      = 'present',
 ) {
 
-  include firewalld::reload
+  $_args = delete_undef_values({
+    'ensure'           => $ensure,
+    'short'            => $short,
+    'description'      => $description,
+    'ports'            => $port,
+    'modules'          => $module,
+    'ipv4_destination' => $destination.dig('ipv4'),
+    'ipv6_destination' => $destination.dig('ipv6'),
+  })
 
-  # Service files may only contain alphanumeric characters and underscores.
-  # This is not documented, but has been experimentally confirmed.
+  # Remove legacy files so that we don't end up with conflicts
```

Comment:
Are there going to be any legacy files anymore?

Reply:
@alexjfisher Honestly, I don't know and couldn't think of a fool-proof way to figure it out. I'll put a note in that the next release of the module will remove that code segment if that works for you.